chainreg/chainregistry.go

```diff
@@ -431,6 +442,20 @@ func NewChainControl(cfg *Config) (*ChainControl, error) {
 				return nil, err
 			}
 		}
+
+		// Bitcoind uses a widely-used lock for its GetBestBlock
+		// endpoint which makes it unsuitable for use as a health
+		// check because it can block for a long time. Instead, we use
+		// a call to the uptime endpoint as our health check function.
+		chainConn, err := rpcclient.New(rpcConfig, nil)
+		if err != nil {
+			return nil, err
+		}
+		cc.HealthCheck = func() error {
+			_, err := chainConn.RawRequest("uptime", nil)
```

Comment:
Which version of `bitcoind` was this call added with? Perhaps we should do a version check on start up to warn the user about the feature, possibly falling back to the older way of doing things?

Reply:
First added in [0.15](https://github.com/bitcoin/bitcoin/blob/master/doc/release-notes/release-notes-0.15.0.md), will add a version check 👍

Reply:
Discussed online, Segwit added in [0.16](https://github.com/bitcoin/bitcoin/blob/master/doc/release-notes/release-notes-0.16.0.md) so you can't run lnd with a version that doesn't have the `uptime` endpoint, version check not required.